### ISSUE TITLE
JS file always generated even when no entry defined

### ISSUE THREAD

**Issue by tedw** [roots/bud#1416]
### Agreement

- [X] This is not a duplicate of an existing issue
- [X] I have read the [guidelines for Contributing to Roots Projects](https://github.com/roots/.github/blob/master/CONTRIBUTING.md)
- [X] This is not a personal support request that should be posted on the [Roots Discourse](https://discourse.roots.io/) community

### Expected Behavior

No JS file should be generated when there’s only a single Sass file listed in `entry()`

### Actual Behavior

A JS file is generated with the following contents:

```js
"use strict";(self.webpackChunksage=self.webpackChunksage||[]).push([[328],{828:function(){}},function(s){s(s.s=828)}]);
```

### Steps To Reproduce

1. Add a single Sass file to `bud.config.js` using the multiple entrypoint syntax:
   ```js
   bud.entry({
     app: ['@scripts/app', '@styles/app.scss'],
     admin: '@styles/admin.scss',
   })
   ```
1. Run `yarn build`
1. Check `/web/wp-content/themes/[theme-name]/public` for `admin.js`

### version

5.8.3

### What package manager are you using?

yarn classic

### version

1.22.10

### Are you using pnpm?

- [ ] yes
- [X] no

### Configuration

```js
/**
 * @typedef {import('@roots/bud').Bud} bud
 *
 * @param {bud} app
 */
module.exports = async (app) => {
  // Add import-glob
  // “setLoader” tells Bud where the loader is located
  // “setItem” tells Bud the loader’s name
  // “setRule” specifies the rule you want Bud to use when evaluating the modules
  // https://discourse.roots.io/t/use-external-compilers-in-bud/22406/13
  app.build
    .setLoader('import-glob', require.resolve('import-glob'))
    .setItem('import-glob', { loader: 'import-glob' })
    .setRule('import-glob', {
      test: /\.(js|s?[ca]ss)$/,
      enforce: 'pre',
    });

  // Include the import-glob rule in the list of sass loaders
  app.build.rules.sass.setUse(items => [...items, 'import-glob']);

  // Postcss plugins
  // https://bud.js.org/extensions/bud-postcss
  app.postcss
    .setPlugin('postcss-preset-env', [
      require.resolve('postcss-preset-env'),
      // Add CSS var fallbacks for old browsers
      { preserve: true },
    ])
    .setPlugin('postcss-custom-properties', [
      require.resolve('postcss-custom-properties'),
      {
        // Tell PostCSS to not alter the following selectors
        // https://github.com/csstools/postcss-plugins/blob/main/plugin-packs/postcss-preset-env/FEATURES.md
        features: {
          'any-link-pseudo-class': false,
          'focus-visible-pseudo-class': false,
          'focus-within-pseudo-class': false,
          'has-pseudo-class': false,
        }
      }
    ]);

  app.postcss.setPlugin('postcss-safe-parser', require.resolve('postcss-safe-parser'));

  // There is an upstream issue (with eslint-webpack-plugin) which causes the linter to show no errors after the initial compilation, regardless of any violations.
  // Until this is resolved, you should use memory caching.
  // https://github.com/roots/bud/releases/tag/v5.8.0
  app.persist('memory');

  app
    /**
     * Application entrypoints
     *
     * Paths are relative to your resources directory
     */
    .entry({
      app: ['@scripts/main', '@styles/main.scss'],
      gutenberg: ['@scripts/gutenberg', '@styles/gutenberg.scss'],
      // FIXME: Figure out why admin.js and login.js are also generated
      admin: '@styles/admin.scss',
      login: '@styles/login.scss',
    })

    /**
     * These files should be processed as part of the build
     * even if they are not explicitly imported in application assets.
     */
    .assets('images')

    /**
     * These files will trigger a full page reload
     * when modified.
     */
    .watch('resources/views/**/*', 'app/**/*')

    /**
     * Target URL to be proxied by the dev server.
     *
     * This should be the URL you use to visit your local development server.
     */
    .proxy('http://[redacted].lndo.site')

    /**
     * Development URL to be used in the browser.
     */
    .serve('http://localhost:3000');
};
```

### Relevant .budfiles

Partial contents of `.budfiles/bud/profile.json`:

```json
"jsconfig.json": {
  "name": "jsconfig.json",
  "path": "/Users/[redacted]/web/wp-content/themes/trevor-mexico/jsconfig.json",
  "module": {
    "compilerOptions": {
      "allowSyntheticDefaultImports": true,
      "jsx": "preserve",
      "lib": [
        "dom",
        "dom.iterable",
        "esnext"
      ],
      "module": "commonjs",
      "moduleResolution": "node",
      "paths": {
        "@scripts/*": [
          "./resources/scripts/*"
        ],
        "@styles/*": [
          "./resources/styles/*"
        ]
      }
    },
    "exclude": [
      "./public",
      "./node_modules"
    ]
  }
},
```

Partial contents of `.budfiles/bud/webpack.config.js`:

```js
module.exports = {
  "entry": {
    "app": {
      "import": [
        "@scripts/main",
        "@styles/main.scss"
      ]
    },
    "gutenberg": {
      "import": [
        "@scripts/gutenberg",
        "@styles/gutenberg.scss"
      ]
    },
    "admin": {
      "import": [
        "@styles/admin.scss"
      ]
    },
    "login": {
      "import": [
        "@styles/login.scss"
      ]
    }
  },
  "bail": true,
  "cache": {
    "type": "memory"
  },
  ```

**Comment by tedw** [comment]
Not sure if it’s related, but `enqueueCss()` doesn’t seem to work in the following cases:

```php
add_action('login_enqueue_scripts', function() {
    bundle('login')->enqueueCss();
});
```
```php
add_action('enqueue_block_editor_assets', function() {
    bundle('gutenberg')->enqueueCss();
}, 100);
```

Using `enqueue()` still works, so it’s not a big deal, but would like to know if this is expected or not. If not, I’m happy to submit another issue if you like.

Thanks for all the hard work on Sage 10! We just upgraded from 9 and things have been working well so far, aside from this minor issue 😉.

**Comment by kellymears** [comment]
Thanks for the report and the encouragement. The above PR will fix the problem. 

Since bud output works when using `enqueue` it seems like the issue with `bundle` would be best made in the [`roots/acorn` repo](https://github.com/roots/acorn).

**Comment by tedw** [comment]
@kellymears Thanks so much for the quick reply and PR! I’ll drop a note about `enqueuCss` in the Acorn repo (sorry, forgot that’s not a Bud thing).

**Comment by tedw** [comment]
@kellymears It seems this fix caused a new issue. I didn’t notice until today, but when I enqueue a custom CSS file on the login page I now get this error locally when the dev server isn’t running, as well as in production. It works fine when `bud dev` is running, though.

<img width="691" src="https://user-images.githubusercontent.com/864716/171279350-4a6a3c51-c5ac-45d1-8b6e-5aff178f11c8.png">

Here’s the code from `wp-content/themes/[theme-name]/app/setup.php:19` it’s complaining about:

```php
add_action('login_enqueue_scripts', function() {
    bundle('login')->enqueue();// <- this is the line
});
```

Here’s my entry config:

```js
bud.entry({
  app: ['@scripts/main', '@styles/main.scss'],
  login: '@styles/login.scss'
})
```

Downgrading to Bud 5.8.3 fixes the issue, so I’m pretty sure it’s related to that fix to remove empty JS files (https://github.com/roots/bud/pull/1417). FWIW I tested in WP 5.9 and 6.0 and there was no difference.

I also tried using `enqueueCss()` instead of `enqueue()` but that still doesn’t seem to be working (I just created an issue for that https://github.com/roots/acorn/issues/229). I wonder if that’s the actual root cause here 🤔 

Thanks in advance for the help!

**Update:** I can also get it to work if I add a fake JS file in `entrypoints.json`:

```json
{
  "app": {
    "css": [
      "app.12b8fd.css"
    ],
    "js": [
      "app.689995.js"
    ],
    "dependencies": []
  },
  "login": {
    "css": [
      "login.8d6319.css"
    ],
    "js": [
      "fake.js"
    ],
    "dependencies": []
  }
}
```
